analyses/malawi/flooding_trigger/mwi-trigger.md

```diff
@@ -0,0 +1,222 @@
+        val = 10*np.round(f_rp(year) / 10)
+        rp_dict[station][year] = val
+
+df_rps = pd.DataFrame(rp_dict)
+```
+
+### Overview of historical discharge
+
+```python
+# Return periods to focus on, with display colours
+rps = {
+    3: '#32a852',
+    5: '#9c2788'
+}
+
+for station in STATIONS: 
+    da_plt = da_glofas_reanalysis[station].sel(time=slice('1999-01-01','2020-12-31'))
+    df_flood_mvac = df_mvac_flood_ta[df_mvac_flood_ta['name']==station]
+
+    fig, ax = plt.subplots()
+    da_plt.plot(x='time', add_legend=True, ax=ax)
+    ax.set_title(f'Historical streamflow at {station}')
+    ax.set_xlabel("Date")
+    ax.set_ylabel('Discharge [m$^3$ s$^{-1}$]')
+
+    for i in range(0,len(df_floodscan_event['start_date'])):
+        ax.axvspan(np.datetime64(df_floodscan_event['start_date'][i]), np.datetime64(df_floodscan_event['end_date'][i]), alpha=0.5, color='#FE5E1E')
+    
+    for key, value in rps.items():
+        ax.axhline(rp_dict[station][key], 0, 1, color=value, label=f'{str(key)} return period')
+        
+    ax.legend()
+    
+    plt.savefig(PLOT_DIR / f'{station}_historical_discharge_glofas_overview_rps.png')
+```
+
+### Identifying glofas events
+
+```python
+def get_groups_above_threshold(observations, threshold):
+    return np.where(np.diff(np.hstack(([False],
+                                           observations > threshold,
+                                           [False]))))[0].reshape(-1, 2)
+
+def get_detection_stats(df_glofas, df_impact, buffer=True):
+    TP = 0 
+    FP = 0
+    tot_events = len(df_impact.index)
+    tot_activations = len(df_glofas.index)
+
+    for index, row in df_glofas.iterrows():
+
+        TP_ = False
+        act_dates = np.array(pd.date_range(row['start_date'], row['end_date']))
+
+        for index,row in df_impact.iterrows():
+            if buffer:
+                event_dates = np.array(pd.date_range(row['start_date_buffer'], row['end_date_buffer']))
+            else:
+                event_dates = np.array(pd.date_range(row['start_date'], row['end_date']))
+            if (set(act_dates) & set(event_dates)):
+                TP+=1
+                TP_ = True
+                df_impact = df_impact.drop([index,])
+                
+
+        if not TP_:
+            FP+=1
+
+    FN = tot_events - TP 
+    
+    return TP, FP, FN
+
+def get_more_stats(TP, FP, FN):
+    precision = TP / (TP + FP)
+    recall = TP / (TP + FN)
+    f1 = 2 / ((1/recall) + (1/precision))
+    return precision, recall, f1
+```
+
+```python
+# Select the station and desired return period
+
+for station in STATIONS:
+    
+    dur = 3
+    detection_stats = {}
+
+    for rp in df_rps.index:
+
+        rp_stats = {}
+
+        # Get the indices from where the threshold would have been met
+        vals = da_glofas_reanalysis[station].values
+        groups = get_groups_above_threshold(vals, df_rps.at[rp, station])
+        groups = [group for group in groups if group[1] - group[0] >= dur]
```

Comment:
I like your idea of putting everything in a data frame. Since you're doing so, you coud remove this line and then select later based on `num_days`, I think that's much clearer